### ISSUE TITLE
[BUG] Pet Sated Debuff not removed at all

### ISSUE THREAD

**Issue by TheDunds** [JamminL/wotlk-classic-bugs#365]
#### 1.1. Description
Hunter Pet sated debuff is not getting removed on encounter end at all
With the new implemented Sated Debuff and the Cooldown resetting mechanics implemented for wotlk there is this new issue upon encounter end.
  #### 1.2. How to Reproduce
           1. Get the stated debuff on a boss fight 
           2. Notice pet stated debuff is not removed upon enconter end at all 
            #### 1.3  Source Material

#### 2.1. Description
Sated debuff to be removed on encouter end like all other players 
A sudo workaround would be to have a second and even third same pet and call stable it out, This should be fixed as having to call a stabled pet and having mutiple of the same pet in your stable is not user friendly.

**Comment by TheDunds** [comment]
This bug persists even upon killing the boss and the encounter ending.
Having to have 2 or 3 of the same pet to have a work around for the exhaustion debuff is not user friendly.
Anyone not willing to use there pet stabled slots on 2 or 3 of the same pet will be at a disadvantage compared to other users who will give up the quality of life of having more pet slots. 
The exhaustion debuff does tick down on a stabled pet from further testing.

**Comment by TheDunds** [comment]
Debuff remains on pet even upon pet death

**Comment by TheDunds** [comment]
updated

**Comment by TheDunds** [comment]
On PTR Patch 3.4.1.47245 I went into kara with a hunter and shaman, I Engaged Attumen the Huntsman casted Hero/Bloodlust the full duration hero went then both the hunter the pet and shaman died, upon reentering kara and reviving pet the sated debuff was removed.


I Then went and made 2 2v2 arena teams consisting of hunter shaman on both sides 
The pets still had debuffs on them when i left arena 
I joined a new game and the debuff was removed. I believe this is the intended way as previously pet sated was not resting on a new game entered.

This bug seems to be fixed.
